static/js/stream_edit.js

```diff
@@ -488,8 +506,12 @@ exports.initialize = function () {
         var stream_id = get_stream_id(e.target);
         var stream = stream_data.get_sub_by_id(stream_id);
         var template_data = {
-            is_private: stream.can_make_public,
             stream_id: stream_id,
+            stream_name: stream.name,
+            is_public: !stream.can_make_public,
+            is_private: stream.can_make_public && !stream.history_public_to_subscribers,
+            is_private_with_public_history: (stream.can_make_public &&
+                                             stream.history_public_to_subscribers),
```

Comment:
This code was wrong; I think it should have been obvious that it was wrong, because if you read it, the things on the RHS didn't line up with the RHS.  I fixed it with the below change:

```
diff --git a/static/js/stream_edit.js b/static/js/stream_edit.js
index 10ce91b..665fde1 100644
--- a/static/js/stream_edit.js
+++ b/static/js/stream_edit.js
@@ -508,9 +508,9 @@ exports.initialize = function () {
         var template_data = {
             stream_id: stream_id,
             stream_name: stream.name,
-            is_public: !stream.can_make_public,
-            is_private: stream.can_make_public && !stream.history_public_to_subscribers,
-            is_private_with_public_history: (stream.can_make_public &&
+            is_public: !stream.invite_only,
+            is_private: stream.invite_only && !stream.history_public_to_subscribers,
+            is_private_with_public_history: (stream.invite_only &&
                                              stream.history_public_to_subscribers),
         };
         var change_privacy_modal = templates.render("subscription_stream_privacy_modal", template_data);
```

Your version seemed to work, but was at best a landmine for future changes.